commit 9ffa2d5dbcc5f110e36832888bfd6574d78d5f27
Author: Akira Kakuto <kakuto@fuk.kindai.ac.jp>
Date:   Mon Jun 11 06:26:38 2018 +0000

    Remove xfree(file_name) in order to avoid to output garbage.
    
    git-svn-id: svn://tug.org/texlive/trunk@47983 c570f23f-e606-0410-a88d-b1316a301751

diff --git a/Build/source/texk/web2c/pdftexdir/ChangeLog b/Build/source/texk/web2c/pdftexdir/ChangeLog
index d276de6c4d5..13ab1e6e062 100644
--- a/Build/source/texk/web2c/pdftexdir/ChangeLog
+++ b/Build/source/texk/web2c/pdftexdir/ChangeLog
@@ -1,3 +1,11 @@
+2018-06-11  Akira Kakuto  <kakuto@fuk.kindai.ac.jp>
+
+	* pdftoepdf.cc: Remove xfree(file_name) in order to avoid
+	to output garbage.
+	* pdftoepdf-newpoppler.cc: Remove _WIN32 changes, since
+	utf-8 names are transformed to Unicode and _wopened
+	in poppler.
+ 
 2018-06-10  TANAKA Takuji  <ttk@t-lab.opal.ne.jp>
 
 	* ptexlib.h, pdftoepdf.cc, pdftoepdf-newpoppler.cc:
diff --git a/Build/source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc b/Build/source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc
index 77f537810a6..750579d6135 100644
--- a/Build/source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc
+++ b/Build/source/texk/web2c/pdftexdir/pdftoepdf-newpoppler.cc
@@ -147,15 +147,6 @@ static XRef *xref = 0;
 static PdfDocument *find_add_document(char *file_name)
 {
     PdfDocument *p = pdfDocuments;
-#ifdef _WIN32
-    wchar_t *fnamew;
-    if (file_system_codepage == CP_UTF8 && win32_codepage != CP_UTF8) {
-        fnamew = get_wstring_from_mbstring(CP_UTF8, file_name, fnamew=NULL);
-        xfree(file_name);
-        file_name = get_mbstring_from_wstring(win32_codepage, fnamew, file_name=NULL);
-        xfree(fnamew);
-    }
-#endif
     while (p && strcmp(p->file_name, file_name) != 0)
         p = p->next;
     if (p) {
diff --git a/Build/source/texk/web2c/pdftexdir/pdftoepdf.cc b/Build/source/texk/web2c/pdftexdir/pdftoepdf.cc
index 747c6838c00..eba6448ce77 100644
--- a/Build/source/texk/web2c/pdftexdir/pdftoepdf.cc
+++ b/Build/source/texk/web2c/pdftexdir/pdftoepdf.cc
@@ -170,7 +170,6 @@ static PdfDocument *find_add_document(char *file_name)
     wchar_t *fnamew;
     if (file_system_codepage == CP_UTF8 && win32_codepage != CP_UTF8) {
         fnamew = get_wstring_from_mbstring(CP_UTF8, file_name, fnamew=NULL);
-        xfree(file_name);
         file_name = get_mbstring_from_wstring(win32_codepage, fnamew, file_name=NULL);
         xfree(fnamew);
     }
